x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/inference/RerankExec.java

```diff
@@ -102,22 +102,15 @@ public UnaryExec replaceChild(PhysicalPlan newChild) {
 
     @Override
     public List<Attribute> output() {
-        if (planHasAttribute(child(), scoreAttribute)) {
-            return child().output();
+        if (lazyOutput == null) {
+            lazyOutput = mergeOutputAttributes(List.of(scoreAttribute), child().output());
         }
-
-        return mergeOutputAttributes(List.of(scoreAttribute), child().output());
+        return lazyOutput;
     }
 
     @Override
     protected AttributeSet computeReferences() {
-        AttributeSet.Builder refs = Rerank.computeReferences(rerankFields).asBuilder();
-
-        if (planHasAttribute(child(), scoreAttribute)) {
-            refs.add(scoreAttribute);
-        }
-
-        return refs.build();
+        return Rerank.computeReferences(rerankFields);
```

Comment:
I _think_ we need to add the scoreAttribute here as well the way we had it before.
I am just looking at what we do in the Analyzer when we resolve the score attribute.
If the child plan has the score attribute, we reuse it.